pkg/miscellaneous/miscellaneous.go

```diff
@@ -462,5 +464,39 @@ func GetInitialClusterState(ctx context.Context, logger logrus.Entry, clientSet
 		return ClusterStateExisting
 	}
 
-	return ClusterStateNew
+	return ""
+}
+
+// DoPromoteMember promotes a given learner to a voting member.
+func DoPromoteMember(ctx context.Context, member *etcdserverpb.Member, cli etcdClient.ClusterCloser, logger *logrus.Logger) error {
+	memPromoteCtx, cancel := context.WithTimeout(ctx, brtypes.DefaultEtcdConnectionTimeout)
+	defer cancel()
+
+	//Member promote call will succeed only if member is in sync with leader, and will error out otherwise
+	_, err := cli.MemberPromote(memPromoteCtx, member.ID)
+	if err == nil {
+		//Member successfully promoted
+		logger.Infof("Member %v with [ID: %v] has been promoted", member.GetName(), strconv.FormatUint(member.GetID(), 16))
+		return nil
+	} else if errored.Is(err, rpctypes.Error(rpctypes.ErrGRPCMemberNotLearner)) {
+		//Member is not a learner
+		logger.Info("Member ", member.Name, " : ", member.ID, " already a voting member of cluster.")
+		return nil
+	}
+	return err
+}
+
+// CheckLearner checks whether a learner(non-voting) member present or not.
+func CheckLearner(ctx context.Context, cli etcdClient.ClusterCloser) (bool, error) {
```

Comment:
Change name to `CheckForLearner` or `CheckIfLearnerPresent`. Maybe also return the learner's member ID, if required later on.